explainability-core/src/test/java/org/kie/trustyai/explainability/local/counterfactual/CounterfactualExplainerTest.java

```diff
@@ -146,10 +151,16 @@ void testNonEmptyInput(int seed) throws ExecutionException, InterruptedException
 
     @ParameterizedTest
     @ValueSource(ints = { 0, 1, 2 })
+    @Timeout(value = 2, unit = TimeUnit.MINUTES, threadMode = Timeout.ThreadMode.SEPARATE_THREAD)
     void testCounterfactualMatch(int seed) throws ExecutionException, InterruptedException, TimeoutException {
+
+        System.out.println("Print something BEFORE seeding #####");
```

Comment:
Are these prints critical or can we remove them?

Reply:
This PR is not ready yet :)
needs a few clean up

Reply:
Done :)